src/lib/corchuelo.rs

```diff
@@ -30,50 +30,86 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-use std::collections::VecDeque;
+use std::hash::{Hash, Hasher};
+use std::cmp::Ordering;
 
 use cactus::Cactus;
 use cfgrammar::{Symbol, TIdx};
 use lrlex::Lexeme;
 use lrtable::{Action, StIdx};
 use num_traits::{PrimInt, Unsigned};
 
+use astar::dijkstra;
+use mf::apply_repairs;
 use parser::{Node, Parser, ParseRepair, Recoverer};
 
 const PARSE_AT_LEAST: usize = 3; // N in Corchuelo et al.
 const PORTION_THRESHOLD: usize = 10; // N_t in Corchuelo et al.
-const INSERT_THRESHOLD: usize = 4; // N_i in Corchuelo et al.
-const DELETE_THRESHOLD: usize = 3; // N_d in Corchuelo et al.
+const TRY_PARSE_AT_MOST: usize = 250;
```

Comment:
Is there a reason for this number?

Reply:
It's the same as in MF but, yes, it's pulled out of my rear end. Basically once we've found different repair sequences, we see which one of them leads to the longest successful parse. However, this is somewhat slow, so we don't want to parse until the end of the file, so we only try `TRY_PARSE_AT_MOST` tokens. I chose a relatively high static value because I couldn't really work out how to make it a dynamic value.

Reply:
I wasn't expecting "my rear end" as an answer but fair enough! :D